neptunecontrib/create_experiment_comparison_comment.py

```diff
@@ -0,0 +1,211 @@
+# limitations under the License.
+#
+
+"""Create a markdown file with an experiment comparison table.
+
+Get a diff between experimenst across metrics, parameters, and properties and save it to a file as a markdown table.
+
+Attributes:
+    tag_ids(list(str)): UUID tags of experiments you would like to compare. It works only for 2 experiments.
+        You can pass it either as --tag_ids or -i. For example, --tag_ids a892ee0ds 09asajd902.
+    experiment_ids(list(str)): Experiment ids of experiments you would like to compare. It works only for 2 experiments.
+        You can pass it either as --experiment_ids or -e. For example, --experiment_ids GIT-83 GIT-82.
+    api_token(str): Neptune api token. If you have NEPTUNE_API_TOKEN environment
+        variable set to your API token you can skip this parameter. You can pass it either as --neptune_api_token or -t.
+    project_name(str): Full name of the project. E.g. "neptune-ai/neptune-examples",
+        If you have PROJECT_NAME environment variable set to your Neptune project you can skip this parameter.
+        You can pass it either as --project_name or -p.
+    filepath(str): filepath of the output markdown file. You can pass it either as --filepath or -f.
+
+Example:
+    Create a file, comparison.md, with a comparison table of experiments GIT-83 and GIT-82::
+
+        $ python -m neptunecontrib.create_experiment_comparison_comment \
+            --tag_ids a892ee0ds 09asajd902 \
+            --api_token ANONYMOUS \
+            --project_name shared/neptune-actions \
+            --filepath comment_body.md
+
+Note:
+    If you keep your neptune api token in the NEPTUNE_API_TOKEN environment variable
+    you can skip the --api_token.
+    If you keep your full neptune project name in the PROJECT_NAME environment variable
+    you can skip the --project_name.
+
+"""
+
+import argparse
+
+import neptune
+import numpy as np
+
+
+def get_experiment_data_by_id(arguments):
+    project = neptune.init(project_qualified_name=arguments.project_name,
+                           api_token=arguments.api_token)
+    return project.get_leaderboard(id=arguments.experiment_ids)
+
+
+def get_experiment_data_by_tag(arguments):
+    project = neptune.init(project_qualified_name=arguments.project_name,
+                           api_token=arguments.api_token)
+
+    experiment_ids = []
+    for tag in arguments.tag_ids:
+        exp = project.get_experiments(tag=tag)[0]
+        experiment_ids.append(exp.id)
+
+    return project.get_leaderboard(id=experiment_ids)
+
+
+def find_experiment_diff(df):
+    selected_cols, cleaned_cols = [], []
+    for col in df.columns:
+        for name in ['channel_', 'parameter_', 'property_']:
+            if name in col and not any(excluded_name in col for excluded_name in ['stderr', 'stdout']):
+                selected_cols.append(col)
+                cleaned_cols.append(col.replace(name, ''))
+    df = df[['id'] + selected_cols]
```

Comment:
change name of assigned variable - shadowing is bad practice

Reply:
ok